src/engine/Settings.cs

```diff
@@ -174,6 +174,17 @@ private Settings()
     /// </summary>
     public SettingValue<bool> CheatsEnabled { get; set; } = new SettingValue<bool>(false);
 
+    /// <summary>
+    ///   If false username will be set to System username
+    /// </summary>
+    public SettingValue<bool> CustomUsernameEnabled { get; set; } = new SettingValue<bool>(false);
+
+    /// <summary>
+    ///   Username that the user can choose
+    ///   Settings this to null will crash on save
+    /// </summary>
+    public SettingValue<string> CustomUsername { get; set; } = new SettingValue<string>(string.Empty);
```

Comment:
Where does the crash happen if this is null? I think it might be a good idea to try to fix that.